pkg/iac/scanners/terraform/parser/evaluator.go

```diff
@@ -296,6 +307,8 @@ func (e *evaluator) expandBlockForEaches(blocks terraform.Blocks, isDynamic bool
 		forEachVal := forEachAttr.Value()
 
 		if forEachVal.IsNull() || !forEachVal.IsKnown() || !forEachAttr.IsIterable() {
+			e.debug.Log(`Failed to expand block %q. Invalid "for-each" argument: %s. Must be known and iterable.`,
```

Comment:
In addition to this, should we also do a further check to make sure that the terraform configuration we just parsed, is valid? This is in reference to https://github.com/aquasecurity/trivy/issues/7099#issuecomment-2214674673

In the above case, I think we should we also log the fact that we purposely ignored an invalid terraform configuration file.

Reply:
What do you mean by valid configuration? Syntactically or valid for evaluation? In the first case we return an error, in the second case I think the notion of invalid does not fit here, rather incomplete. We may lack some data to evaluate individual blocks and dependencies, not the whole configuration. This can happen mostly because of missing input data or using data blocks or resources that are not static (e.g. random provider).

Reply:
> What do you mean by valid configuration? 

I meant this case: https://github.com/aquasecurity/trivy/issues/7099#issuecomment-2213060444

Reply:
Added logging if the user has not set the variable https://github.com/aquasecurity/trivy/pull/7133/commits/1febd97f3ec3e8ec79b8e0d8b36ba35f5e6b30e7